docs: re-add `AsyncAPI Documents` and change code snippet (#2274)
asyncapi/website: pages/docs/tutorials/getting-started/asyncapi-documents.md

---
title: "AsyncAPI documents"
date: 2019-04-01T10:56:52+01:00
menu:
  docs:
    parent: 'getting-started'
weight: 101
---

An AsyncAPI document is a file that defines and annotates the different components of **a specific Event-Driven API**.

> The format of the file must be JSON or YAML; however, only the subset of YAML that matches the JSON capabilities is allowed.

Reviewer: ```suggestion
The file format must be JSON or YAML; however, only the subset of YAML that matches the JSON capabilities is allowed.
```

<CodeBlock>
{`asyncapi: 3.0.0
info:
  title: Example application
  version: '0.1.0'
channels:
  userSignup:
    address: 'user/signedup'
    messages:
      userSignedupMessage: 
        description: A message describing that a user just signed up.
        payload:
          type: object
          additionalProperties: false
          properties:
            fullName:
              type: string
            email:
              type: string
              format: email
            age:
              type: integer
              minimum: 18
operations:
  publishUserSignedup:
    action: 'send'
    channel:
      $ref: '#/channels/userSignup'`}
</CodeBlock>

> The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive to the broker.

Reviewer: ```suggestion
The AsyncAPI document is a machine-readable definition of your Event-Driven API. That document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive at the broker.
```

> Your API documentation is now machine-readable –easily parseable by code— so the myriad of useful applications is endless.

Reviewer: ```suggestion
Your API documentation is now machine-readable (easily parseable by code) so the myriad of useful applications is endless.
```

---
title: "AsyncAPI documents"
date: 2019-04-01T10:56:52+01:00
menu:
  docs:
    parent: 'getting-started'
weight: 101
---

An AsyncAPI document is a file that defines and annotates the different components of **a specific Event-Driven API**.

The format of the file must be JSON or YAML; however, only the subset of YAML that matches the JSON capabilities is allowed.

<CodeBlock>
{`asyncapi: 3.0.0
info:
  title: Example application
  version: '0.1.0'

channels:
  userSignup:
    address: 'user/signedup'
    messages:
      userSignedupMessage: 
        description: A message describing that a user just signed up.
        payload:
          type: object
          additionalProperties: false
          properties:
            fullName:
              type: string
            email:
              type: string
              format: email
            age:
              type: integer
              minimum: 18

operations:
  publishUserSignedup:
    action: 'send'
    channel:
      $ref: '#/channels/userSignup'`}

> </CodeBlock>

Reviewer 2: ```suggestion
<CodeBlock>
{`asyncapi: 3.0.0
info:
  title: Example application
  version: '0.1.0'
channels:
  userSignup:
    address: 'user/signedup'
    messages:
      userSignedupMessage: 
        description: A message describing that a user just signed up.
        payload:
          type: object
          additionalProperties: false
          properties:
            fullName:
              type: string
            email:
              type: string
              format: email
            age:
              type: integer
              minimum: 18
operations:
  publishUserSignedup:
    action: 'send'
    channel:
      $ref: '#/channels/userSignup'`}
</CodeBlock>
```

small change, build fails as CodeBlock do not like empty spaces

Author: > small change, build fails as CodeBlock do not like empty spaces

hmm...new thing to learn :thinking: 

resolved with a03a7bb

> The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages your application receives, and even apply API management policies to your messages before they arrive to your broker.

Reviewer 2: ```suggestion
The AsyncAPI document is a machine-readable definition of your Event-Driven API. This document can be used afterward to generate documentation and code, validate the messages that `Example application` sends, and even apply API management policies to your messages before they arrive to the broker.
```

Author: resolved with 24fdfd8
